petergreve/week2/Ass3/index.html

```diff
@@ -0,0 +1,233 @@
+					    					key={i}
+					    					handleInput={that.handleInput}
+					    					archiveMessage={that.archiveMessage}
+					    					createMessage={that.createMessage}
+					    					deleteMessage={that.deleteMessage}
+					    					messages={this.filterMessages(list.listId)}
+					    					archived={this.filterArchived(list.listId)}
+					    					moveMessage={that.moveMessage}
+					    					list={list} />;
+					   		},this)}
+					    </ol>			
+					</div>
+		},
+		handleInput: function(message) {
+			if (message.length<200) {
+				this.setState({
+		          errorMessage: '',
+		          allMessages: this.state.allMessages.concat([message]),
+		      	});
+			} else {
+				this.setState({
+		          errorMessage: message,
+				});
+			}
+		},
+		createList: function() {
+			var newList = [{
+				listId: this.state.listCount,
+				listName: this.newListName.value
+			}];
+
+			this.setState({
+		          lists: this.state.lists.concat(newList),
+		          listCount: this.state.listCount + 1
+		      	});
+		},
+		createMessage: function(text, listId) {
+			if (text.length<200) {
+				var newMessage = [{
+					messageId: this.state.messageCount,
+					text: text,
+					isArchived: false,
+					listId: listId
+				}];
+
+				this.setState({
+			          allMessages: this.state.allMessages.concat(newMessage),
+			          messageCount: this.state.messageCount + 1
+			      	});
+			} 
+		},
+		filterMessages: function(listId) {
+			return this.state.allMessages.filter((m) => {
+				return m.isArchived == false && m.listId == listId;
+			});
+		},
+		filterArchived: function(listId) {
+			return this.state.allMessages.filter((m) => {
+				return m.isArchived == true && m.listId == listId;
+			});
+		},
+		deleteMessage: function(messageId) {
+			var newAllMessages = this.state.allMessages.filter((m) => {
+				return m.messageId != messageId;
+			});
+
+			this.setState({
+				allMessages: newAllMessages
+			});
+		},
+		archiveMessage: function(messageId, isArchived) {
+			var archivedMessage = this.state.allMessages.filter((m) => {
+				return m.messageId == messageId;
+			})[0];
+
+
+			archivedMessage.isArchived = isArchived;
+
+			var newAllMessages = this.state.allMessages.filter((m) => {
+				return m.messageId != messageId;
+			});
+
+			this.setState({
+				allMessages: newAllMessages.concat([archivedMessage])
+			});
+		},
+		moveMessage: function(messageId, down) {
+			var archivedMessage = this.state.allMessages.filter((m) => {
+				return m.messageId == messageId;
+			})[0];
+
+			if (down && archivedMessage.listId < this.state.listCount - 1)
+				archivedMessage.listId = archivedMessage.listId + 1;
+			else if (!down && archivedMessage.listId > 0)
+				archivedMessage.listId = archivedMessage.listId - 1;
+			else return
+
+			var newAllMessages = this.state.allMessages.filter((m) => {
+				return m.messageId != messageId;
+			});
+
+			this.setState({
+				allMessages: newAllMessages.concat([archivedMessage])
```

Comment:
I get why you are doing this, but you are in fact still manipulating state when you set the `listId`. But we will talk about that today